http-client/src/main/java/io/airlift/http/client/jetty/DefaultHttpClientLogger.java

```diff
@@ -118,4 +130,46 @@ private static void recoverTempFiles(String logPath)
             }
         }
     }
+
+    private static class FlushingFileAppender<T>
+            extends RollingFileAppender<T>
+    {
+        private final AtomicLong lastFlushed = new AtomicLong(System.nanoTime());
+        private final long flushIntervalNanos;
+
+        private FlushingFileAppender(Duration flushInterval)
+        {
+            this.flushIntervalNanos = flushInterval.roundTo(NANOSECONDS);
+        }
+
+        @Override
+        protected void subAppend(T event)
+        {
+            super.subAppend(event);
+
+            long now = System.nanoTime();
+            long last = lastFlushed.get();
+            if (((now - last) > flushIntervalNanos) && lastFlushed.compareAndSet(last, now)) {
+                flush();
+            }
+        }
+
+        @SuppressWarnings("Duplicates")
+        private void flush()
```

Comment:
I feel like we are going to be getting extra flushes, since we don't know when the super class flushes.